sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/partition_client.hpp

```diff
@@ -89,32 +80,54 @@ namespace Azure { namespace Messaging { namespace EventHubs {
 
     /** @brief Closes the connection to the Event Hub service.
      */
-    void Close()
-    {
-      for (size_t i = 0; i < m_receivers.size(); i++)
-      {
-        m_receivers[i].Close();
-      }
-    }
+    void Close() { m_receiver.Close(); }
+
+  private:
+    friend class _detail::PartitionClientFactory;
+    /// The message receiver used to receive events from the partition.
+    Azure::Core::Amqp::_internal::MessageReceiver m_receiver;
+
+    /// The name of the offset to start receiving events from.
+    //    std::string m_offsetExpression;
+
+    /// The options used to create the PartitionClient.
+    PartitionClientOptions m_partitionOptions;
+
+    /// The name of the partition.
+    //    std::string m_partitionId;
+
+    /** @brief RetryOptions controls how many times we should retry an operation in
+     * response to being throttled or encountering a transient error.
+     */
+    Azure::Core::Http::Policies::RetryOptions m_retryOptions{};
+
+    // Azure::Core::Amqp::Common::_internal::AsyncOperationQueue<
+    //    Azure::Core::Amqp::Models::AmqpMessage,
+    //    Azure::Core::Amqp::Models::_internal::AmqpError>
+    //    m_receivedMessageQueue;
 
     /** Creates a new PartitionClient
      *
-     * @param options The options used to create the PartitionClient.
-     * @param retryOptions The retry options used to create the PartitionClient.
-     *
+     * @param messageReceiver Message Receiver for the partition client.
+     * @param options options used to create the PartitionClient.
+     * @param retryOptions controls how many times we should retry an operation in response to being
+     * throttled or encountering a transient error.
      */
     PartitionClient(
+        Azure::Core::Amqp::_internal::MessageReceiver const& messageReceiver,
         PartitionClientOptions options,
-        Azure::Core::Http::Policies::RetryOptions retryOptions)
-    {
-      m_partitionOptions = options;
-      RetryOptions = retryOptions;
-    }
-
-    /// @brief Push the message receiver back to the vector of receivers.
-    void PushBackReceiver(Azure::Core::Amqp::_internal::MessageReceiver& receiver)
-    {
-      m_receivers.push_back(std::move(receiver));
-    }
+        Azure::Core::Http::Policies::RetryOptions retryOptions);
```

Comment:
Should `PartitionClientOptions` derive from `ClientOptions`? We are adding `RetryOptions` from there, but what happens if that list grows?

Reply:
`ClientOptions` is inextricably linked to HTTP transports. 

The only information in it which isn't 100% tied to HTTP are the `RetryPolicyOptions` (mostly) and `TelemetryOptions` (theoretically - not 100% sure what has to happen to tracing provider for tracing and metric support for AMQP).

We could consider refactoring ClientOptions to be transport agnostic but I'm not sure it's worth the churn.